modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractJavaExtensionModelLoader.java

```diff
@@ -119,16 +103,35 @@ protected void declareExtension(ExtensionLoadingContext context) {
     delegateFactory.apply(extensionType, version).declare(context);
   }
 
+  private Collection<DeclarationEnricher> getPrivilegedDeclarationEnrichers(ExtensionLoadingContext context) {
+    Class<?> extensionType = getExtensionType(context);
+    String declarationEnricherName = DeclarationEnrichers.class.getName();
+    try {
+      Class annotation = context.getExtensionClassLoader().loadClass(declarationEnricherName);
+      DeclarationEnrichers enrichers = extensionType.getAnnotation((Class<DeclarationEnrichers>) annotation);
+      return enrichers != null ? stream(enrichers.value()).map(this::instantiateOrFail).collect(toList()) : emptyList();
```

Comment:
this should happen using the extension's classloader, otherwise you might not be able to instantiate the enrichers